services/wallet/activity/filter.go

```diff
@@ -53,31 +56,81 @@ type TokenCode string
 // nil means all
 // see allTokensFilter and noTokensFilter
 type Tokens struct {
-	Assets       []TokenCode          `json:"assets"`
-	Collectibles []eth_common.Address `json:"collectibles"`
-	EnabledTypes []TokenType          `json:"enabledTypes"`
+	Assets       []TokenCode   `json:"assets"`
+	Collectibles []eth.Address `json:"collectibles"`
+	EnabledTypes []TokenType   `json:"enabledTypes"`
 }
 
 func noAssetsFilter() Tokens {
-	return Tokens{[]TokenCode{}, []eth_common.Address{}, []TokenType{CollectiblesTT}}
+	return Tokens{[]TokenCode{}, []eth.Address{}, []TokenType{CollectiblesTT}}
 }
 
 func allTokensFilter() Tokens {
 	return Tokens{}
 }
 
-func allAddressesFilter() []eth_common.Address {
-	return []eth_common.Address{}
+func allAddressesFilter() []eth.Address {
+	return []eth.Address{}
 }
 
 func allNetworksFilter() []common.ChainID {
 	return []common.ChainID{}
 }
 
 type Filter struct {
-	Period                Period               `json:"period"`
-	Types                 []Type               `json:"types"`
-	Statuses              []Status             `json:"statuses"`
-	Tokens                Tokens               `json:"tokens"`
-	CounterpartyAddresses []eth_common.Address `json:"counterpartyAddresses"`
+	Period                Period        `json:"period"`
+	Types                 []Type        `json:"types"`
+	Statuses              []Status      `json:"statuses"`
+	Tokens                Tokens        `json:"tokens"`
+	CounterpartyAddresses []eth.Address `json:"counterpartyAddresses"`
+}
+
+// TODO: consider sorting by saved address and contacts to offload the client from doing it at runtime
```

Comment:
Not sure about this if it makes sense yet, didn't check how we resolve names in the frontend but if rules are manageable in SQL this might be the efficient way.